# ASM NGS Challenge 2015

The [ASM NGS Challenge 2015][] was part of the [1st ASM Conference on Rapid Next-Generation Sequencing and Bioinformatic Pipelines for Enhanced Molecular Epidemiologic Investigation of Pathogens][] which provided interested participants an opporunity to compare software and analysis methods for classifying isolates from a real-world outbreak scenerio using next-generation sequencing data.  Participants were provided with with sequencing data along with minimial epidemiological information, and were asked to classify mixtures of clinical,environmental, and food isolates as matches or non-matches.

SNVPhyl (version `0.2-beta-1`) was entered into the challenge and was used to construct phylogenetic trees and SNV distance matrices to aid in classifying the isolates.

| *Listeria monocytogenes*        | *Salmonella* Enteritidis        |
|:-------------------------------:|:-------------------------------:|
| [![lm-phylo-small][]][lm-phylo] | [![se-phylo-small][]][se-phylo] |

The results can be found at <https://github.com/apetkau/snvphyl-asm-ngs-2015>.

[ASM NGS Challenge 2015]: https://github.com/katholt/NGSchallenge
[1st ASM Conference on Rapid Next-Generation Sequencing and Bioinformatic Pipelines for Enhanced Molecular Epidemiologic Investigation of Pathogens]: http://conferences.asm.org/index.php/2012-02-09-21-04-52/past-conferences/2-uncategorised/341-conference-scope-2015-ngs
[lm-phylo-small]: images/lm-phylo-small.png
[se-phylo-small]: images/se-phylo-small.png
[lm-phylo]: images/lm-phylo.png
[se-phylo]: images/se-phylo.png
